server/src/main/java/org/elasticsearch/action/admin/indices/refresh/RefreshReplicationRequest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.indices.refresh;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.action.support.replication.ReplicationRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.tasks.TaskId;
+
+import java.io.IOException;
+
+public class RefreshReplicationRequest extends ReplicationRequest<RefreshReplicationRequest> {
```

Comment:
This is very similar to what was originally proposed in https://github.com/elastic/elasticsearch/pull/92818/files#diff-a13bf6f0242976703d5312291cea9c0c08d9c8ad12c667cc79ae6212ff2c0a37 but we ended up scrapping when we made Refresh manually send actions to unpromotable shards. Now there's a single request for both promotable & unpromotable shards again.

Note we could extend `TransportReplicationAction`'s generics to have a separate Request class for unpromotable shards (rather than sharing the same ReplicationRequest for both promotable & unpromotable shards). I actually tried to go down that way, and although it works, it made the code more complex. At which point I thought there's no need to have a separate class for now.